### PR TITLE
Addressing Archboard Feedback

### PR DESCRIPTION
**Pull request by yogeshmo**
# Description

The SDK is now model-less. All tests and samples have been updated to reflect this change.

AAD Authentication support was added.

Minor changes for the most recent architecture board review were added in as well.

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-email/azure/communication/email/_generated/aio/operations/_patch.py [hunk 1]
```diff
@@ -6,59 +6,71 @@
...
+JSON = MutableMapping[str, Any]  # pylint: disable=unsubscriptable-object
 
 class EmailOperations(EmailOperationsGenerated):
 
-    def __return_message_id(self, pipeline_response, _, response_headers):
+    def _return_message_id(self, pipeline_response, _, response_headers):
         return response_headers['x-ms-request-id']
 
-    async def send(
+    async def send(  # pylint: disable=inconsistent-return-statements
```
Comment (annatisch) [inline sdk/communication/azure-communication-email/azure/communication/email/_generated/aio/operations/_patch.py:24]: I'm not sure if you also need to patch the generated overloads - though maybe it won't matter given that you have wrapped the generated client.

File: sdk/communication/azure-communication-email/azure/communication/email/_generated/aio/operations/_patch.py [hunk 2]
```diff
@@ -6,59 +6,71 @@
...
+         Default value is None.
         :paramtype content_type: str
-        :return: SendEmailResult or the result of cls(response)
-        :rtype: ~azure.communication.email.models.SendEmailResult
+        :return: JSON object
+        :rtype: JSON
         :raises ~azure.core.exceptions.HttpResponseError:
         """
+
         message_id = await super().send(
```
Comment (annatisch) [inline sdk/communication/azure-communication-email/azure/communication/email/_generated/aio/operations/_patch.py:60]: Is there any way to recover the message ID if the connection is lost before this value is returned?

File: sdk/communication/azure-communication-email/azure/communication/email/_shared/utils.py [hunk 3]
```diff
@@ -9,12 +9,10 @@
 import calendar
 from typing import (cast,
                     Tuple,
-                    Union,
                     )
 from datetime import datetime
 from msrest.serialization import TZ_UTC
-from azure.core.credentials import AccessToken
-
+from azure.core.credentials import AccessToken, AzureKeyCredential
```
Comment (annatisch) [inline sdk/communication/azure-communication-email/azure/communication/email/_shared/utils.py:15]: The changes to the _shared directory will need to be synced across the other communications SDKs

File: sdk/communication/azure-communication-email/azure/communication/email/_shared/utils.py [hunk 4]
```diff
@@ -126,7 +127,7 @@ def get_authentication_policy(
         from azure.core.pipeline.policies import BearerTokenCredentialPolicy
         return BearerTokenCredentialPolicy(
             credential, "https://communication.azure.com//.default")
-    if isinstance(credential, str):
+    if isinstance(credential, AzureKeyCredential):
```
Comment (annatisch) [inline sdk/communication/azure-communication-email/azure/communication/email/_shared/utils.py:130]: Is this breaking to any existing support for `str` across other Communications SDKs?

File: sdk/communication/azure-communication-email/azure/communication/email/aio/_email_client_async.py [hunk 5]
```diff
@@ -77,16 +81,86 @@ def from_connection_string(
     @distributed_trace_async
     async def send(
         self,
-        email_message: EmailMessage,
+        email_message: JSON,
```
Comment (annatisch) [inline sdk/communication/azure-communication-email/azure/communication/email/aio/_email_client_async.py:84]: In the generated code we support `Union[JSON, IO]` as input - is there any reason you removed the IO support from the convenience layer?

File: sdk/communication/azure-communication-email/azure/communication/email/_email_client.py [hunk 6]
```diff
@@ -48,11 +53,10 @@ def __init__(
 
         self._api_version = kwargs.pop("api_version", DEFAULT_VERSION)
 
-        authentication_policy = HMACCredentialsPolicy(endpoint, credential)
+        authentication_policy = get_authentication_policy(endpoint, credential)
```
Comment (annatisch) [inline sdk/communication/azure-communication-email/azure/communication/email/_email_client.py:56]: It seems that this `_email_client.py` file (and the async equivalent) is pretty redundant now, as we could easily move these pieces into the `_patch.py` files. Though you could leave that refactor for the next preview if you prefer.

File: sdk/communication/azure-communication-email/azure/communication/email/_generated/aio/operations/_patch.py (hunk 2 above)
Comment (yogeshmo) [reply sdk/communication/azure-communication-email/azure/communication/email/_generated/aio/operations/_patch.py:60]: There's no way to do this, the message id is returned by the service when that call is made.

File: sdk/communication/azure-communication-email/azure/communication/email/_shared/utils.py (hunk 4 above)
Comment (yogeshmo) [reply sdk/communication/azure-communication-email/azure/communication/email/_shared/utils.py:130]: Good point, I'll make sure this works across SDKs.

File: sdk/communication/azure-communication-email/azure/communication/email/aio/_email_client_async.py (hunk 5 above)
Comment (yogeshmo) [reply sdk/communication/azure-communication-email/azure/communication/email/aio/_email_client_async.py:84]: No specific reason. I'll add this in the wrapper.

File: sdk/communication/azure-communication-email/azure/communication/email/_email_client.py (hunk 6 above)
Comment (yogeshmo) [reply sdk/communication/azure-communication-email/azure/communication/email/_email_client.py:56]: Gotcha. Will leave this refactor for the next preview.

**Review by petrsvihlik — CHANGES_REQUESTED** [review]
Good to see the support for `AzureKeyCredential`! 

I'm taking a note for the ACS Identity team to fully onboard it in the Python SDK (https://skype.visualstudio.com/SPOOL/_workitems/edit/3067523)

**Review by petrsvihlik — APPROVED** [review]

**Review by AikoBB — APPROVED** [review]

**Pull request merged**